Model/Api.php

```diff
@@ -1526,6 +1554,118 @@ private function createCache()
+            && !boolval($this->getIntegrationSetting('parallel_offline', false))
+        ) {
+            $this->logger->debug('Parallel processing disabled for non-realtime.');
+
+            return;
+        }
+
+        if (
+            $this->imported
+            && !boolval($this->getIntegrationSetting('parallel_import', false))
+        ) {
+            $this->logger->debug('Parallel processing disabled for imports.');
+
+            return;
+        }
+
+        if (!function_exists('pcntl_fork')) {
+            $this->logger->debug('Parallel processing not possible because PCNTL module is not installed.');
+
+            return;
+        }
+
+        // Update the list of campaigns for this contact, and get a list of those that need to be processed.
+        $campaignsFound = false;
+        foreach (array_keys($this->campaignModel->getLeadCampaigns($this->contact, true)) as $campaignId) {
+            if (!isset($this->contactCampaigns[$campaignId])) {
+                $this->contactCampaigns[$campaignId] = false;
+                $campaignsFound                      = true;
+            }
+        }
+        if (!$campaignsFound) {
+            $this->logger->debug('No campaigns to process in parallel');
+
+            return;
+        }
+
+        // Commit any MySQL changes and close the connection/s to prepare for a process fork.
+        $connection = $this->em->getConnection();
+        if ($connection->isConnected()) {
+            // Complete any active transactions.
+            if ($connection->isTransactionActive()) {
+                $connection->commit();
+            }
+            // Close the connection.
+            $connection->close();
+        }
+
+        // Calling this way since this is a soft dependency, it's also faster in 7+
+        $pid = call_user_func('pcntl_fork');
+        if (-1 === $pid) {
+            $this->logger->error(
+                'Contact Source '.
+                ($this->contactSource ? $this->contactSource->getId() : 'NA').
+                ': Could not fork the process'
+            );
+
+            return;
+        } elseif ($pid) {
+            $this->logger->debug('Parent continues process.');
+
+            // Parent process can continue.
+            return;
+        } else {
+            // Child process has work to do.
+            foreach ($this->contactCampaigns as $campaignId => &$isProcessed) {
+                if (!$isProcessed) {
+                    $campaign = $this->campaignModel->getEntity($campaignId);
+                    if ($campaign && $campaign->getIsPublished()) {
+                        $this->logger->info('Child kicking off campaign: '.$campaignId);
+
+                        $this->campaignExecutioner->execute($campaign, [$this->contact->getId()]);
+                    }
+                    $isProcessed = true;
+                }
+            }
+            $this->logger->debug('Parallel process complete.');
+            exit;
+        }
+    }
+
     /**
```

Comment:
Doesn't the parent pricess need to wait around until the children all finish? ... `pcntl_waitpid`

Reply:
I specifically do not want to wait. There's no information I need from the children during the  request at runtime.